server/lib/schema.js

```diff
@@ -23,7 +23,8 @@ exports.testPage = Joi.object().keys({
   test: Joi.array().required().min(2).includes(Joi.object().required().keys({
     title: Joi.string().required().trim().min(1).max(255),
     defer: Joi.string().default('n').valid('y', 'n'),
-    code: Joi.string().required().trim().min(1).max(defaults.mediumTextLength)
+    code: Joi.string().required().trim().min(1).max(defaults.mediumTextLength),
+    id: Joi.number().integer() // optional. only present when editing
```

Comment:
when editing a test, its `id` is also present in the form submission so need to allow it in the schema here